integration-client/src/main/java/com/lightbend/lagom/javadsl/client/integration/LagomClientFactory.java

```diff
@@ -0,0 +1,262 @@
+/*
+ * Copyright (C) 2016 Lightbend Inc. <http://www.lightbend.com>
+ */
+package com.lightbend.lagom.javadsl.client.integration;
+
+import akka.actor.ActorSystem;
+import akka.japi.Effect;
+import akka.stream.ActorMaterializer;
+import akka.stream.Materializer;
+import com.lightbend.lagom.internal.client.*;
+import com.lightbend.lagom.internal.registry.ServiceRegistry;
+import com.lightbend.lagom.internal.registry.ServiceRegistryServiceLocator;
+import com.lightbend.lagom.javadsl.api.ServiceInfo;
+import com.lightbend.lagom.javadsl.api.ServiceLocator;
+import com.lightbend.lagom.javadsl.jackson.JacksonExceptionSerializer;
+import com.lightbend.lagom.javadsl.jackson.JacksonSerializerFactory;
+import io.netty.channel.EventLoopGroup;
+import io.netty.channel.nio.NioEventLoopGroup;
+import org.asynchttpclient.DefaultAsyncHttpClientConfig;
+import org.pcollections.PVector;
+import org.pcollections.TreePVector;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import play.api.Configuration;
+import play.api.Environment;
+import play.api.Mode;
+import play.api.inject.ApplicationLifecycle;
+import play.api.libs.ws.WSClient;
+import play.api.libs.ws.WSClientConfig;
+import play.api.libs.ws.WSConfigParser;
+import play.api.libs.ws.ahc.AhcConfigBuilder;
+import play.api.libs.ws.ahc.AhcWSClient;
+import play.api.libs.ws.ahc.AhcWSClientConfig;
+import play.api.libs.ws.ahc.AhcWSClientConfigParser;
+import play.api.libs.ws.ssl.SystemConfiguration;
+import scala.Function0;
+import scala.concurrent.Future;
+
+import java.io.Closeable;
+import java.io.File;
+import java.net.URI;
+import java.util.Collection;
+import java.util.Optional;
+import java.util.concurrent.Callable;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.function.Function;
+
+/**
+ * Factory for creating Lagom service clients.
+ *
+ * This is designed for use from non Lagom systems.
+ *
+ * Generally, there should be only one instance of this per system.  Additionally, since it holds thread and connection
```

Comment:
Because there may be times when you have two separate lifecycles, eg two applications sharing the same classloader, each would be responsible for starting and shutting down their own client factory.  If we enforced only one, it would prevent that. Another example is running tests, you may start many up in tests run concurrently, each test starting it and shutting it down when it's finished, there's nothing wrong with doing that.  Finally, the only way to enforce it is global state, ie sticking it in a static var.  That's not something we'd really want to do.